provider/pkg/schema/gen.go

```diff
@@ -1092,6 +1092,27 @@ func (ctx *context) propertyTypeSpec(parentName string, propSchema *jsschema.Sch
 		}
 	}
 
+	var mapType *jsschema.Schema
+	if propSchema.AdditionalProperties != nil {
+		mapType = propSchema.AdditionalProperties.Schema
+	} else if propSchema.PatternProperties != nil && len(propSchema.PatternProperties) == 1 {
+		// TODO: Capture pattern add add to documentation or even provider metadata for early validation feedback.
```

Comment:
Let's file an enhancement ticket for this.

Reply:
Good shout. Added this: https://github.com/pulumi/pulumi-aws-native/issues/1346